deploy/helm/templates/_common-discovery-config.tpl

```diff
@@ -151,13 +151,20 @@ transform/istio-metrics:
         - extract_count_metric(true) where (metric.name == "{{ .Values.otel.metrics.autodiscovery.prefix }}istio_request_duration_milliseconds")
         - set(metric.name, "k8s.istio_request_duration_milliseconds_sum_temp") where metric.name == "{{ .Values.otel.metrics.autodiscovery.prefix }}istio_request_duration_milliseconds_sum"
         - set(metric.name, "k8s.istio_request_duration_milliseconds_count_temp") where metric.name == "{{ .Values.otel.metrics.autodiscovery.prefix }}istio_request_duration_milliseconds_count"
-        - set(resource.attributes["istio"], "true") 
+        - set(resource.attributes["istio"], "true")
 
 transform/istio-metric-datapoints:
   metric_statements:
     - statements:
         - set(datapoint.attributes["dest.sw.server.address.fqdn"], datapoint.attributes["destination_service"]) where metric.name == "{{ .Values.otel.metrics.autodiscovery.prefix }}istio_request_bytes_sum" and IsMatch(datapoint.attributes["destination_service"], "^(https?://)?[a-zA-Z0-9][-a-zA-Z0-9]*\\.[a-zA-Z0-9][-a-zA-Z0-9\\.]*(:\\d+)?$") and not(IsMatch(datapoint.attributes["destination_service"], ".*\\.cluster\\.local$")) and not(IsMatch(datapoint.attributes["destination_service"], "^(https?://)?\\d+\\.\\d+\\.\\d+\\.\\d+(:\\d+)?$"))
 
+transform/istio-relationship-types:
+  metric_statements:
+    - statements:
+        - set(resource.attributes["tcp"], "true") where (metric.name == "{{ .Values.otel.metrics.autodiscovery.prefix }}istio_tcp_sent_bytes_total" or metric.name == "{{ .Values.otel.metrics.autodiscovery.prefix }}istio_tcp_received_bytes_total") and datapoint.attributes["request_protocol"] == "tcp"
```

Comment:
This will need to be updated later to use some kind of a "delta" metric. Maybe it could be replaced by a variation of the following statements, just checking for `request_protocol=="tcp"`, if it works?